react/features/youtube-player/middleware.js

```diff
@@ -0,0 +1,166 @@
+    (conference, { dispatch, getState }, previousConference) => {
+        if (conference && conference !== previousConference) {
+            conference.addCommandListener(SHARED_VIDEO,
+                ({ value, attributes }) => {
+                    const localParticipantId = getLocalParticipant(getState()).id;
+
+                    switch (attributes.state) {
+                    case 'start':
+                        dispatch(setSharedVideoOwner(attributes.from));
+                        dispatch(participantJoined({
+                            conference,
+                            id: value,
+                            isFakeParticipant: true,
+                            avatarURL: `https://img.youtube.com/vi/${value}/0.jpg`,
+                            name: 'YouTube'
+                        }));
+                        dispatch(pinParticipant(value));
+                        break;
+                    case 'playing':
+                        if (localParticipantId !== attributes.from) {
+                            dispatch(setSharedVideoStatus('playing', attributes.time));
+                        }
+                        break;
+                    case 'pause':
+                        if (localParticipantId !== attributes.from) {
+                            dispatch(setSharedVideoStatus('pause', attributes.time));
+                        }
+                        break;
+                    case 'stop':
+                        dispatch(setSharedVideoStatus('stop', ''));
+                        dispatch(participantLeft(value, conference));
+                        break;
+                    }
+                }
+            );
+        }
+    });
+
+
+/**
+ * Dispatches shared video status.
+ *
+ * @param {Store} store - The redux store.
+ * @param {string} next - Todo add doc.
+ * @param {string} action - Todo add doc.
+ * @returns {Function}
+ */
+function _toggleSharedVideo(store, next, action) {
+    const state = store.getState();
+    const { ownerId } = state['features/youtube-player'];
+    const localParticipant = getLocalParticipant(state);
+
+    const conference = getCurrentConference(state);
+    const { status } = state['features/youtube-player'];
+    const fakeParticipant = state['features/base/participants'].filter(p => p.isFakeParticipant)[0];
+
+    if (status === 'playing' || status === 'start' || status === 'pause') {
+        if (ownerId === localParticipant.id) {
+            sendShareVideoCommand(fakeParticipant.id, 'stop', conference);
+        }
+    } else {
+        store.dispatch(showEnterVideoLinkPrompt(id => _onVideoLinkEntered(store, id)));
+    }
+
+    return next(action);
+}
+
+/**
+ * Sends SHARED_VIDEO start command.
+ *
+ * @param {Store} store - The redux store.
+ * @param {string} id - The youtube id of the video to be shared.
+ * @returns {void}
+ */
+function _onVideoLinkEntered(store, id) {
+    const conference = getCurrentConference(store.getState());
```

Comment:
Unlikely, but maybe guard against delayed action processing, so check if conference is not undefined